### ISSUE TITLE
BlueSpice and SMW 

### ISSUE THREAD

**Issue by DestructionOfPlanetJupiter** [SemanticMediaWiki/SemanticMediaWiki#750]
```
    NO_VALID_VALUE mit Datentyp Text (0 Vorkommen)
    NO_VALID_VALUE mit Datentyp Zahl (0 Vorkommen)
    NO_VALID_VALUE mit Datentyp Quellcode (0 Vorkommen)
    NO_VALID_VALUE mit Datentyp Zahl (0 Vorkommen)
    Band (0)
    NO_VALID_VALUE (0)
    NO_VALID_VALUE (0)
    NO_VALID_VALUE (0)
    NO_VALID_VALUE (0)
    NO_VALID_VALUE (0)
    NO_VALID_VALUE mit Datentyp Datum (0 Vorkommen)
    NO_VALID_VALUE (0)
    NO_VALID_VALUE (0)
    NO_VALID_VALUE (0)
    NO_VALID_VALUE (0)
    NO_VALID_VALUE (0)
    NO_VALID_VALUE (0)
    Herkunft (0)
    NO_VALID_VALUE (0)
    NO_VALID_VALUE mit Datentyp Wahrheitswert (0 Vorkommen)
    NO_VALID_VALUE mit Datentyp Seite (0 Vorkommen)
    NO_VALID_VALUE (0)
    NO_VALID_VALUE (0)
    NO_VALID_VALUE mit Datentyp Datum (0 Vorkommen)
``` 
This is how it looks like for me once SMW 2.1 and BlueSpice 2.23.0. I tried to create some properties, but as can be seen they do not contain any content and cannot be used in any kind of way. Ask and show appear to be quite pointless in such a scenario. 

I have set the namespace to 3000, as has been suggested on Sourceforge, but also to no avail. How can this be dealt with? Is there a fix or a setting with which SMW and BS can be used?

**Comment by mwjames** [comment]
I'm guessing you did run `rebuildData`?

> NO_VALID_VALUE 

??

> SMW 2.1 and BlueSpice 2.23.0

No idea what BlueSpice is or does

> I have set the namespace to 3000, as has been suggested on Sourceforge

@kghbln 3000 as to why, any idea?

**Comment by DestructionOfPlanetJupiter** [comment]
The data is there. Once you disable BS all the content is there again. The error message comes out the SMW extension and not as I thought first from BlueSpice.

This would be BlueSpice:
http://sourceforge.net/projects/bluespice/

It is a large extension to MW and adds a lot of other things to it. With it you have more elements at your proposal.

**Comment by mwjames** [comment]
What you can do to narrow down the issue is to run `composer phpunit` from the SMW root directory (depends on MediaWiki containing a test directory) to see whether all tests pass on your system (normally run for a "vanilla" MediaWiki installation [0] without any extra extensions).

[0] https://travis-ci.org/SemanticMediaWiki/SemanticMediaWiki

**Comment by DestructionOfPlanetJupiter** [comment]
Configuration read from /var/www/w/rurakblue/extensions/SemanticMediaWiki/phpunit.xml.dist

..................FF.........................................   61 / 3071 (  1%)
..................................................F..........  122 / 3071 (  3%)
.................F..FFFF.....................................  183 / 3071 (  5%)
.............................................................  244 / 3071 (  7%)
.............................................................  305 / 3071 (  9%)
..........................................F..............F...  366 / 3071 ( 11%)
.............................................................  427 / 3071 ( 13%)
F............................................................  488 / 3071 ( 15%)
.............................................................  549 / 3071 ( 17%)
......................................F.F....................  610 / 3071 ( 19%)
.............................................................  671 / 3071 ( 21%)
.............................................................  732 / 3071 ( 23%)
..............F....................PHP Fatal error:  Class 'SMWLanguageEn' not found in /var/www/w/rurakblue/extensions/SemanticMediaWiki/includes/NamespaceManager.php on line 176
PHP Stack trace:
PHP   1. {main}() /var/www/w/rurakblue/tests/phpunit/phpunit.php:0
PHP   2. PHPUnit_TextUI_Command::main() /var/www/w/rurakblue/tests/phpunit/phpunit.php:233
PHP   3. PHPUnit_TextUI_Command->run() /var/www/w/rurakblue/vendor/phpunit/phpunit/src/TextUI/Command.php:103
PHP   4. PHPUnit_TextUI_TestRunner->doRun() /var/www/w/rurakblue/vendor/phpunit/phpunit/src/TextUI/Command.php:151
PHP   5. PHPUnit_Framework_TestSuite->run() /var/www/w/rurakblue/vendor/phpunit/phpunit/src/TextUI/TestRunner.php:388
PHP   6. PHPUnit_Framework_TestSuite->run() /var/www/w/rurakblue/vendor/phpunit/phpunit/src/Framework/TestSuite.php:716
PHP   7. PHPUnit_Framework_TestSuite->run() /var/www/w/rurakblue/vendor/phpunit/phpunit/src/Framework/TestSuite.php:716
PHP   8. PHPUnit_Framework_TestCase->run() /var/www/w/rurakblue/vendor/phpunit/phpunit/src/Framework/TestSuite.php:716
PHP   9. PHPUnit_Framework_TestResult->run() /var/www/w/rurakblue/vendor/phpunit/phpunit/src/Framework/TestCase.php:687
PHP  10. PHPUnit_Framework_TestCase->runBare() /var/www/w/rurakblue/vendor/phpunit/phpunit/src/Framework/TestResult.php:609
PHP  11. PHPUnit_Framework_TestCase->runTest() /var/www/w/rurakblue/vendor/phpunit/phpunit/src/Framework/TestCase.php:731
PHP  12. ReflectionMethod->invokeArgs() /var/www/w/rurakblue/vendor/phpunit/phpunit/src/Framework/TestCase.php:846
PHP  13. SMW\Test\NamespaceManagerTest->testExecution() /var/www/w/rurakblue/vendor/phpunit/phpunit/src/Framework/TestCase.php:846
PHP  14. SMW\NamespaceManager->run() /var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/includes/NamespaceManagerTest.php:64
PHP  15. SMW\NamespaceManager->initContentLanguage() /var/www/w/rurakblue/extensions/SemanticMediaWiki/includes/NamespaceManager.php:38
Script php ../../tests/phpunit/phpunit.php -c phpunit.xml.dist handling the phpunit event returned with an error



  [RuntimeException]
  Error Output: PHP Fatal error:  Class 'SMWLanguageEn' not found in /var/www
  /w/rurakblue/extensions/SemanticMediaWiki/includes/NamespaceManager.php on
  line 176
  PHP Stack trace:
  PHP   1. {main}() /var/www/w/rurakblue/tests/phpunit/phpunit.php:0
  PHP   2. PHPUnit_TextUI_Command::main() /var/www/w/rurakblue/tests/phpunit/
  phpunit.php:233
  PHP   3. PHPUnit_TextUI_Command->run() /var/www/w/rurakblue/vendor/phpunit/
  phpunit/src/TextUI/Command.php:103
  PHP   4. PHPUnit_TextUI_TestRunner->doRun() /var/www/w/rurakblue/vendor/php
  unit/phpunit/src/TextUI/Command.php:151
  PHP   5. PHPUnit_Framework_TestSuite->run() /var/www/w/rurakblue/vendor/php
  unit/phpunit/src/TextUI/TestRunner.php:388
  PHP   6. PHPUnit_Framework_TestSuite->run() /var/www/w/rurakblue/vendor/php
  unit/phpunit/src/Framework/TestSuite.php:716
  PHP   7. PHPUnit_Framework_TestSuite->run() /var/www/w/rurakblue/vendor/php
  unit/phpunit/src/Framework/TestSuite.php:716
  PHP   8. PHPUnit_Framework_TestCase->run() /var/www/w/rurakblue/vendor/phpu
  nit/phpunit/src/Framework/TestSuite.php:716
  PHP   9. PHPUnit_Framework_TestResult->run() /var/www/w/rurakblue/vendor/ph
  punit/phpunit/src/Framework/TestCase.php:687
  PHP  10. PHPUnit_Framework_TestCase->runBare() /var/www/w/rurakblue/vendor/
  phpunit/phpunit/src/Framework/TestResult.php:609
  PHP  11. PHPUnit_Framework_TestCase->runTest() /var/www/w/rurakblue/vendor/
  phpunit/phpunit/src/Framework/TestCase.php:731
  PHP  12. ReflectionMethod->invokeArgs() /var/www/w/rurakblue/vendor/phpunit
  /phpunit/src/Framework/TestCase.php:846
  PHP  13. SMW\Test\NamespaceManagerTest->testExecution() /var/www/w/rurakblu
  e/vendor/phpunit/phpunit/src/Framework/TestCase.php:846
  PHP  14. SMW\NamespaceManager->run() /var/www/w/rurakblue/extensions/Semant
  icMediaWiki/tests/phpunit/includes/NamespaceManagerTest.php:64
  PHP  15. SMW\NamespaceManager->initContentLanguage() /var/www/w/rurakblue/e
  xtensions/SemanticMediaWiki/includes/NamespaceManager.php:38

**Comment by mwjames** [comment]
> Error Output: PHP Fatal error: Class 'SMWLanguageEn' not found in /var/www
/w/rurakblue/extensions/SemanticMediaWiki/includes/NamespaceManager.php on

This only appears to happen when BS is active but without BS being active, all tests pass, right?

**Comment by DestructionOfPlanetJupiter** [comment]
Nope ...

**Comment by mwjames** [comment]
Then I have no idea why on your system [0] can not find the language class that is supposed to be placed in `./SemanticMediaWiki/languages/SMW_LanguageEn.php`

[0] https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/includes/NamespaceManager.php#L181

**Comment by kghbln** [comment]
>  namespace 3000 as to why, any idea?

In general it is recommended to use the namespace numbers 3000+ for [custom namspaces](https://www.mediawiki.org/wiki/Extension_default_namespaces#ID_3000.2B) They add a custom namespace at 1500 for their integrated blog so I do not know where the 3000 comes from. Probably they recommend this for additional custom namespaces, which would be according to recommendation.

Afaik, the people at Hallo Welt who develop BlueSpice do not use SMW at all. So they probably and understandably do not really care if BlueSpice plays with SMW.

**Comment by mwjames** [comment]
> Nope ... same error ... :/
> ..................FF......................................... 61 / 3071 ( 1%)
..................................................F.......... 122 / 3071 ( 3%)
.................F..FFFF.....................................

I was curious about the apparent 'F' (failures) and they are caused by having `$wgLanguageCode` not set to `en` which ties in with the not yet resolved #154 (Make tests user-language agnostic). I'm guessing your LocalSettings to include something like `$wgLanguageCode = de` which still doesn't explain `SMWLanguageEn` exception.

**Comment by DestructionOfPlanetJupiter** [comment]
The thing with a Namespace 3000 derives from this thread:
http://sourceforge.net/p/bluespice/discussion/1249668/thread/905b569b/
I thought it had been worth a try.
======================================================
Correct, $wgLanguageCode = "de"; appears in the LocalSettings.php

I have changed the settings for this variable on a test-wiki to "en" and this would be the result:

Configuration read from /var/www/w/rurakxml2/extensions/SemanticMediaWiki/phpunit.xml.dist

.............................................................   61 / 3071 (  1%)
.............................................................  122 / 3071 (  3%)
.............................................................  183 / 3071 (  5%)
.............................................................  244 / 3071 (  7%)
.............................................................  305 / 3071 (  9%)
.............................................................  366 / 3071 ( 11%)
.............................................................  427 / 3071 ( 13%)
.............................................................  488 / 3071 ( 15%)
.............................................................  549 / 3071 ( 17%)
.............................................................  610 / 3071 ( 19%)
.............................................................  671 / 3071 ( 21%)
.............................................................  732 / 3071 ( 23%)
.............................................................  793 / 3071 ( 25%)
.............................................................  854 / 3071 ( 27%)
.............................................................  915 / 3071 ( 29%)
.............................................................  976 / 3071 ( 31%)
............................................................. 1037 / 3071 ( 33%)
............................................................. 1098 / 3071 ( 35%)
............................................................. 1159 / 3071 ( 37%)
............................................................. 1220 / 3071 ( 39%)
............................................................. 1281 / 3071 ( 41%)
............................................................. 1342 / 3071 ( 43%)
............................................................. 1403 / 3071 ( 45%)
............................................................. 1464 / 3071 ( 47%)
............................................................. 1525 / 3071 ( 49%)
............................................................. 1586 / 3071 ( 51%)
............................................................. 1647 / 3071 ( 53%)
............................................................. 1708 / 3071 ( 55%)
............................................................. 1769 / 3071 ( 57%)
............................................................. 1830 / 3071 ( 59%)
............................................................. 1891 / 3071 ( 61%)
............................................................. 1952 / 3071 ( 63%)
............................................................. 2013 / 3071 ( 65%)
............................................................. 2074 / 3071 ( 67%)
............................................................. 2135 / 3071 ( 69%)
............................................................. 2196 / 3071 ( 71%)
............................................................. 2257 / 3071 ( 73%)
............................................................. 2318 / 3071 ( 75%)
............................................................. 2379 / 3071 ( 77%)
............................................................. 2440 / 3071 ( 79%)
............................................................. 2501 / 3071 ( 81%)
............................................................. 2562 / 3071 ( 83%)
............................................................. 2623 / 3071 ( 85%)
.......................................S..................... 2684 / 3071 ( 87%)
.................................................SSSSSSS..... 2745 / 3071 ( 89%)
...............................
SMW\Tests\Integration\MediaWiki\Import\Maintenance\RebuildDataMaintenanceTest testRebuildData ran for 11.071 seconds
.............................. 2806 / 3071 ( 91%)
............................................................. 2867 / 3071 ( 93%)
............................................................. 2928 / 3071 ( 95%)
............................................................. 2989 / 3071 ( 97%)
.SSS......................................................... 3050 / 3071 ( 99%)
.....................

Time: 1.5 minutes, Memory: 182.50Mb

There were 11 skipped tests:

1) SMW\Tests\Integration\MediaWiki\SearchInPageDBIntegrationTest::testSearchForGeographicCoordinateValueAsTerm
Requires 'Geographic coordinate' to be a supported data type (see Semantic Maps)

/var/www/w/rurakxml2/extensions/SemanticMediaWiki/tests/phpunit/Integration/MediaWiki/SearchInPageDBIntegrationTest.php:62
/var/www/w/rurakxml2/extensions/SemanticMediaWiki/tests/phpunit/MwDBaseUnitTestCase.php:104

2) SMW\Tests\Integration\SPARQLStore\QueryResultLookupWithoutBaseStoreIntegrationTest::testQuerySubjects_afterUpdatingSemanticData
Requires a SPARQLStore instance

/var/www/w/rurakxml2/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/QueryResultLookupWithoutBaseStoreIntegrationTest.php:51

3) SMW\Tests\Integration\SPARQLStore\QueryResultLookupWithoutBaseStoreIntegrationTest::testQueryZeroResults_afterSubjectRemoval
Requires a SPARQLStore instance

/var/www/w/rurakxml2/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/QueryResultLookupWithoutBaseStoreIntegrationTest.php:51

4) SMW\Tests\Integration\SPARQLStore\QueryResultLookupWithoutBaseStoreIntegrationTest::testQuerySubjects_onNamspaceRestrictedCondition
Requires a SPARQLStore instance

/var/www/w/rurakxml2/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/QueryResultLookupWithoutBaseStoreIntegrationTest.php:51

5) SMW\Tests\Integration\SPARQLStore\QueryResultLookupWithoutBaseStoreIntegrationTest::testQuerySubobjects_afterUpdatingWithEmptyContainerAllAssociatedEntitiesGetRemovedFromGraph
Requires a SPARQLStore instance

/var/www/w/rurakxml2/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/QueryResultLookupWithoutBaseStoreIntegrationTest.php:51

6) SMW\Tests\Integration\SPARQLStore\RedirectLookupIntegrationTest::testRedirectTragetLookupForNonExistingEntry with data set #0 (SMWExpNsResource Object (...))
Requires a SPARQLStore instance

/var/www/w/rurakxml2/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/RedirectLookupIntegrationTest.php:38

7) SMW\Tests\Integration\SPARQLStore\RedirectLookupIntegrationTest::testRedirectTragetLookupForNonExistingEntry with data set #1 (SMWExpNsResource Object (...))
Requires a SPARQLStore instance

/var/www/w/rurakxml2/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/RedirectLookupIntegrationTest.php:38

8) SMW\Tests\Integration\SPARQLStore\RedirectLookupIntegrationTest::testRedirectTragetLookupForExistingEntry
Requires a SPARQLStore instance

/var/www/w/rurakxml2/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/RedirectLookupIntegrationTest.php:38

9) SMW\Tests\System\InstallationGlobalsProviderIntegrityTest::testNamespaceSettingOnExampleIfSet
Skip test due to missing http://example.org/id/ setting

/var/www/w/rurakxml2/extensions/SemanticMediaWiki/tests/phpunit/System/InstallationConfigurationIntegrityTest.php:60

10) SMW\Tests\System\InstallationGlobalsProviderIntegrityTest::testNamespacesWithSemanticLinksOnTravisCustomNamespace with data set #0 ('GLOBALS', array(true, false, true, false, true, false, true, false, true, false, true, false, true, false, false, false, false, false, true, false, true, false))
Test can only be executed with a specified NS_TRAVIS

/var/www/w/rurakxml2/extensions/SemanticMediaWiki/tests/phpunit/System/InstallationConfigurationIntegrityTest.php:75

11) SMW\Tests\System\InstallationGlobalsProviderIntegrityTest::testNamespacesWithSemanticLinksOnTravisCustomNamespace with data set #1 ('Settings', array(true, false, true, false, true, false, true, false, true, false, true, false, true, false, false, false, false, false, true, false, true, false))
Test can only be executed with a specified NS_TRAVIS

/var/www/w/rurakxml2/extensions/SemanticMediaWiki/tests/phpunit/System/InstallationConfigurationIntegrityTest.php:75

OK, but incomplete, skipped, or risky tests!
Tests: 3071, Assertions: 8829, Skipped: 11.

**Comment by mwjames** [comment]
> Correct, $wgLanguageCode = "de"; appears in the LocalSettings.php

#752 (2.2+) fixed the `wgLanguageCode` dependency so that tests can run language agnostic (meaning independent from the `wgLanguageCode` setting).

> OK, but incomplete, skipped, or risky tests!
Tests: 3071, Assertions: 8829, Skipped: 11.

This is a good result (never mind the skipped tests as they are environment dependent). Are those now run with or without the BS extension enabled?

**Comment by kghbln** [comment]
> The thing with a Namespace 3000 derives from this thread:
http://sourceforge.net/p/bluespice/discussion/1249668/thread/905b569b/
I thought it had been worth a try.

One should really use namespace constants which is recommended since MW 1.10 or so..

**Comment by DestructionOfPlanetJupiter** [comment]
No, a separate wiki ... I just wanted to test it and see what the results would have been. 

the error log of the BS installation is 35KB in size and contains 1000 PHP comments. So ... where can I put it?

**Comment by JeroenDeDauw** [comment]
> the error log of the BS installation is 350KB in size and contains 1000 PHP comments. So ... where can I put it?

pastebin?

> error log of the BS installation is 350KB

![fbi](https://cloud.githubusercontent.com/assets/146040/5856947/c2edaa02-a247-11e4-8bf2-d3be7c71de00.jpg)

**Comment by mwjames** [comment]
```
PHP 21. ExtendedSearch->onArticleSaveComplete() /var/www/w/rurakblue/includes/Hooks.php:206
PHP 22. BuildIndexMainControl->updateIndexWiki() /var/www/w/rurakblue/extensions/BlueSpiceExtensions/ExtendedSearch/ExtendedSearch.class.php:376
PHP 23. BuildIndexMainControl->updateIndexWiki() /var/www/w/rurakblue/extensions/BlueSpiceExtensions/ExtendedSearch/includes/BuildIndex/BuildIndexMainControl.class.php:156
... 
PHP 988. BuildIndexMainControl->updateIndexWiki() /var/www/w/rurakblue/extensions/BlueSpiceExtensions/ExtendedSearch/includes/BuildIndex/BuildIndexMainControl.class.php:156
PHP 989. BsPageContentProvider->getContentFromRevision() /var/www/w/rurakblue/extensions/BlueSpiceExtensions/ExtendedSearch/includes/BuildIndex/BuildIndexMainControl.class.php:151
PHP 990. Parser->preprocess() 
```

I'm particular suspicious about `BuildIndexMainControl->updateIndexWiki()` fired by `ExtendedSearch->onArticleSaveComplete()` which seems to run in circles and only after 996 is reached, SMW is able to catch up with the hook (but probably no the native MW hook, it seems the Parser is invoked by `BlueSpiceFoundation/includes/utility/PageContentProvider.class.php`).

```
PHP 990. Parser->preprocess() /var/www/w/rurakblue/extensions/BlueSpiceFoundation/includes/utility/PageContentProvider.class.php:140
PHP 991. Parser->startParse() /var/www/w/rurakblue/includes/parser/Parser.php:631
PHP 992. Parser->clearState() /var/www/w/rurakblue/includes/parser/Parser.php:5000
PHP 993. Parser->firstCallInit() /var/www/w/rurakblue/includes/parser/Parser.php:291
PHP 994. wfRunHooks() /var/www/w/rurakblue/includes/parser/Parser.php:279
PHP 995. Hooks::run() /var/www/w/rurakblue/includes/GlobalFunctions.php:3995
PHP 996. call_user_func_array:{/var/www/w/rurakblue/includes/Hooks.php:206}() /var/www/w/rurakblue/includes/Hooks.php:206
PHP 997. SMW\MediaWiki\Hooks\HookRegistry->SMW\MediaWiki\Hooks{closure}() 
```

**Comment by DestructionOfPlanetJupiter** [comment]
Yes, that is I why I refrained from the idea of uploading it, because there is actually not much to show, except for a process running in circles.

**Comment by mwjames** [comment]
> Yes, that is I why I refrained from the idea of uploading it, because there is actually not much to show, except for a process running in circles.

In order to narrow down the possibilities of failure, I'd suggest to somehow try to deactivate the `extensions/BlueSpiceExtensions/ExtendedSearch` when running the tests.

**Comment by DestructionOfPlanetJupiter** [comment]
This is what I get with the BlueSpice extension deactivated:
......................................ESE.................... 2623 / 3071 ( 85%)
.......................................S..................... 2684 / 3071 ( 87%)
.................................................SSSSSSS..... 2745 / 3071 ( 89%)
...............................
SMW\Tests\Integration\MediaWiki\Import\Maintenance\RebuildDataMaintenanceTest testRebuildData ran for 14.019 seconds
.............................. 2806 / 3071 ( 91%)
............................................................. 2867 / 3071 ( 93%)
............................................................. 2928 / 3071 ( 95%)
............................................................. 2989 / 3071 ( 97%)
.SSS......................................................... 3050 / 3071 ( 99%)
.....................

Time: 1.75 minutes, Memory: 180.00Mb

There were 2 errors:

1) SMW\Tests\Integration\MediaWiki\Hooks\FileUploadIntegrationTest::testFileUploadForDummyTextFile
fopen(/var/www/w/rurakblue/images/lockdir/4dczolxewj6iay0k3fs98b9lijdau24.lock): failed to open stream: Permission denied

/var/www/w/rurakblue/includes/filebackend/lockmanager/FSLockManager.php:125
/var/www/w/rurakblue/includes/filebackend/lockmanager/FSLockManager.php:72
/var/www/w/rurakblue/includes/filebackend/lockmanager/LockManager.php:202
/var/www/w/rurakblue/includes/filebackend/lockmanager/LockManager.php:111
/var/www/w/rurakblue/includes/filebackend/lockmanager/LockManager.php:93
/var/www/w/rurakblue/includes/filebackend/FileBackend.php:1233
/var/www/w/rurakblue/includes/filerepo/file/LocalFile.php:1869
/var/www/w/rurakblue/includes/filerepo/file/LocalFile.php:1174
/var/www/w/rurakblue/includes/upload/UploadBase.php:738
/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Utils/File/LocalFileUpload.php:86
/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/MediaWiki/Hooks/FileUploadIntegrationTest.php:106
/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/MwDBaseUnitTestCase.php:104

2) SMW\Tests\Integration\MediaWiki\Hooks\FileUploadIntegrationTest::testDummyTextFileUploadForDisabledNamespace
fopen(/var/www/w/rurakblue/images/lockdir/ic35wwtztw7qtm7gbpim542rfuoupj2.lock): failed to open stream: Permission denied

/var/www/w/rurakblue/includes/filebackend/lockmanager/FSLockManager.php:125
/var/www/w/rurakblue/includes/filebackend/lockmanager/FSLockManager.php:72
/var/www/w/rurakblue/includes/filebackend/lockmanager/LockManager.php:202
/var/www/w/rurakblue/includes/filebackend/lockmanager/LockManager.php:111
/var/www/w/rurakblue/includes/filebackend/lockmanager/LockManager.php:93
/var/www/w/rurakblue/includes/filebackend/FileBackend.php:1233
/var/www/w/rurakblue/includes/filerepo/file/LocalFile.php:1869
/var/www/w/rurakblue/includes/filerepo/file/LocalFile.php:1174
/var/www/w/rurakblue/includes/upload/UploadBase.php:738
/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Utils/File/LocalFileUpload.php:86
/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/MediaWiki/Hooks/FileUploadIntegrationTest.php:159
/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/MwDBaseUnitTestCase.php:104

--

There were 12 skipped tests:

1) SMW\Tests\Integration\MediaWiki\Hooks\FileUploadIntegrationTest::testReUploadDummyTextFileToEditFilePage
This test depends on "SMW\Tests\Integration\MediaWiki\Hooks\FileUploadIntegrationTest::testFileUploadForDummyTextFile" to pass.

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/MwDBaseUnitTestCase.php:104

2) SMW\Tests\Integration\MediaWiki\SearchInPageDBIntegrationTest::testSearchForGeographicCoordinateValueAsTerm
Requires 'Geographic coordinate' to be a supported data type (see Semantic Maps)

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/MediaWiki/SearchInPageDBIntegrationTest.php:62
/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/MwDBaseUnitTestCase.php:104

3) SMW\Tests\Integration\SPARQLStore\QueryResultLookupWithoutBaseStoreIntegrationTest::testQuerySubjects_afterUpdatingSemanticData
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/QueryResultLookupWithoutBaseStoreIntegrationTest.php:51

4) SMW\Tests\Integration\SPARQLStore\QueryResultLookupWithoutBaseStoreIntegrationTest::testQueryZeroResults_afterSubjectRemoval
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/QueryResultLookupWithoutBaseStoreIntegrationTest.php:51

5) SMW\Tests\Integration\SPARQLStore\QueryResultLookupWithoutBaseStoreIntegrationTest::testQuerySubjects_onNamspaceRestrictedCondition
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/QueryResultLookupWithoutBaseStoreIntegrationTest.php:51

6) SMW\Tests\Integration\SPARQLStore\QueryResultLookupWithoutBaseStoreIntegrationTest::testQuerySubobjects_afterUpdatingWithEmptyContainerAllAssociatedEntitiesGetRemovedFromGraph
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/QueryResultLookupWithoutBaseStoreIntegrationTest.php:51

7) SMW\Tests\Integration\SPARQLStore\RedirectLookupIntegrationTest::testRedirectTragetLookupForNonExistingEntry with data set #0 (SMWExpNsResource Object (...))
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/RedirectLookupIntegrationTest.php:38

8) SMW\Tests\Integration\SPARQLStore\RedirectLookupIntegrationTest::testRedirectTragetLookupForNonExistingEntry with data set #1 (SMWExpNsResource Object (...))
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/RedirectLookupIntegrationTest.php:38

9) SMW\Tests\Integration\SPARQLStore\RedirectLookupIntegrationTest::testRedirectTragetLookupForExistingEntry
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/RedirectLookupIntegrationTest.php:38

10) SMW\Tests\System\InstallationGlobalsProviderIntegrityTest::testNamespaceSettingOnExampleIfSet
Skip test due to missing http://example.org/id/ setting

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/System/InstallationConfigurationIntegrityTest.php:60

11) SMW\Tests\System\InstallationGlobalsProviderIntegrityTest::testNamespacesWithSemanticLinksOnTravisCustomNamespace with data set #0 ('GLOBALS', array(true, false, true, false, true, false, true, false, true, false, true, false, true, false, false, false, false, false, true, false, true, false))
Test can only be executed with a specified NS_TRAVIS

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/System/InstallationConfigurationIntegrityTest.php:75

12) SMW\Tests\System\InstallationGlobalsProviderIntegrityTest::testNamespacesWithSemanticLinksOnTravisCustomNamespace with data set #1 ('Settings', array(true, false, true, false, true, false, true, false, true, false, true, false, true, false, false, false, false, false, true, false, true, false))
Test can only be executed with a specified NS_TRAVIS

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/System/InstallationConfigurationIntegrityTest.php:75

FAILURES!
Tests: 3070, Assertions: 8797, Errors: 2, Skipped: 12.
Script php ../../tests/phpunit/phpunit.php -c phpunit.xml.dist handling the phpunit event returned with an error

**Comment by DestructionOfPlanetJupiter** [comment]
By the way, there has been a suggestion that gets rid of the problem:
https://www.mediawiki.org/wiki/Thread:Project:Support_desk/This_appears_to_be_a_cross-site_request_forgery;_canceling_save./reply_(19)

And it works ... I have tested it.

**Comment by DestructionOfPlanetJupiter** [comment]
And this is the result for:
BlueSpice activated, additional Namespace settings added (see link above) and the extensions ExtendedSearch and Universal Export deactivated:
......................................ESE.................... 2623 / 3071 ( 85%)
.......................................S..................... 2684 / 3071 ( 87%)
.................................................SSSSSSS..... 2745 / 3071 ( 89%)
...............................
SMW\Tests\Integration\MediaWiki\Import\Maintenance\RebuildDataMaintenanceTest testRebuildData ran for 11.183 seconds
.............................. 2806 / 3071 ( 91%)
............................................................. 2867 / 3071 ( 93%)
............................................................. 2928 / 3071 ( 95%)
............................................................. 2989 / 3071 ( 97%)
.SSS......................................................... 3050 / 3071 ( 99%)
.....................

Time: 1.66 minutes, Memory: 196.00Mb

There were 2 errors:

1) SMW\Tests\Integration\MediaWiki\Hooks\FileUploadIntegrationTest::testFileUploadForDummyTextFile
fopen(/var/www/w/rurakblue/images/lockdir/4dczolxewj6iay0k3fs98b9lijdau24.lock): failed to open stream: Permission denied

/var/www/w/rurakblue/includes/filebackend/lockmanager/FSLockManager.php:125
/var/www/w/rurakblue/includes/filebackend/lockmanager/FSLockManager.php:72
/var/www/w/rurakblue/includes/filebackend/lockmanager/LockManager.php:202
/var/www/w/rurakblue/includes/filebackend/lockmanager/LockManager.php:111
/var/www/w/rurakblue/includes/filebackend/lockmanager/LockManager.php:93
/var/www/w/rurakblue/includes/filebackend/FileBackend.php:1233
/var/www/w/rurakblue/includes/filerepo/file/LocalFile.php:1869
/var/www/w/rurakblue/includes/filerepo/file/LocalFile.php:1174
/var/www/w/rurakblue/includes/upload/UploadBase.php:738
/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Utils/File/LocalFileUpload.php:86
/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/MediaWiki/Hooks/FileUploadIntegrationTest.php:106
/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/MwDBaseUnitTestCase.php:104

2) SMW\Tests\Integration\MediaWiki\Hooks\FileUploadIntegrationTest::testDummyTextFileUploadForDisabledNamespace
fopen(/var/www/w/rurakblue/images/lockdir/ic35wwtztw7qtm7gbpim542rfuoupj2.lock): failed to open stream: Permission denied

/var/www/w/rurakblue/includes/filebackend/lockmanager/FSLockManager.php:125
/var/www/w/rurakblue/includes/filebackend/lockmanager/FSLockManager.php:72
/var/www/w/rurakblue/includes/filebackend/lockmanager/LockManager.php:202
/var/www/w/rurakblue/includes/filebackend/lockmanager/LockManager.php:111
/var/www/w/rurakblue/includes/filebackend/lockmanager/LockManager.php:93
/var/www/w/rurakblue/includes/filebackend/FileBackend.php:1233
/var/www/w/rurakblue/includes/filerepo/file/LocalFile.php:1869
/var/www/w/rurakblue/includes/filerepo/file/LocalFile.php:1174
/var/www/w/rurakblue/includes/upload/UploadBase.php:738
/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Utils/File/LocalFileUpload.php:86
/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/MediaWiki/Hooks/FileUploadIntegrationTest.php:159
/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/MwDBaseUnitTestCase.php:104

--

There were 12 skipped tests:

1) SMW\Tests\Integration\MediaWiki\Hooks\FileUploadIntegrationTest::testReUploadDummyTextFileToEditFilePage
This test depends on "SMW\Tests\Integration\MediaWiki\Hooks\FileUploadIntegrationTest::testFileUploadForDummyTextFile" to pass.

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/MwDBaseUnitTestCase.php:104

2) SMW\Tests\Integration\MediaWiki\SearchInPageDBIntegrationTest::testSearchForGeographicCoordinateValueAsTerm
Requires 'Geographic coordinate' to be a supported data type (see Semantic Maps)

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/MediaWiki/SearchInPageDBIntegrationTest.php:62
/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/MwDBaseUnitTestCase.php:104

3) SMW\Tests\Integration\SPARQLStore\QueryResultLookupWithoutBaseStoreIntegrationTest::testQuerySubjects_afterUpdatingSemanticData
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/QueryResultLookupWithoutBaseStoreIntegrationTest.php:51

4) SMW\Tests\Integration\SPARQLStore\QueryResultLookupWithoutBaseStoreIntegrationTest::testQueryZeroResults_afterSubjectRemoval
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/QueryResultLookupWithoutBaseStoreIntegrationTest.php:51

5) SMW\Tests\Integration\SPARQLStore\QueryResultLookupWithoutBaseStoreIntegrationTest::testQuerySubjects_onNamspaceRestrictedCondition
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/QueryResultLookupWithoutBaseStoreIntegrationTest.php:51

6) SMW\Tests\Integration\SPARQLStore\QueryResultLookupWithoutBaseStoreIntegrationTest::testQuerySubobjects_afterUpdatingWithEmptyContainerAllAssociatedEntitiesGetRemovedFromGraph
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/QueryResultLookupWithoutBaseStoreIntegrationTest.php:51

7) SMW\Tests\Integration\SPARQLStore\RedirectLookupIntegrationTest::testRedirectTragetLookupForNonExistingEntry with data set #0 (SMWExpNsResource Object (...))
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/RedirectLookupIntegrationTest.php:38

8) SMW\Tests\Integration\SPARQLStore\RedirectLookupIntegrationTest::testRedirectTragetLookupForNonExistingEntry with data set #1 (SMWExpNsResource Object (...))
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/RedirectLookupIntegrationTest.php:38

9) SMW\Tests\Integration\SPARQLStore\RedirectLookupIntegrationTest::testRedirectTragetLookupForExistingEntry
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/RedirectLookupIntegrationTest.php:38

10) SMW\Tests\System\InstallationGlobalsProviderIntegrityTest::testNamespaceSettingOnExampleIfSet
Skip test due to missing http://example.org/id/ setting

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/System/InstallationConfigurationIntegrityTest.php:60

11) SMW\Tests\System\InstallationGlobalsProviderIntegrityTest::testNamespacesWithSemanticLinksOnTravisCustomNamespace with data set #0 ('GLOBALS', array(true, false, true, false, true, false, true, false, true, false, true, false, true, false, false, false, false, false, true, false, true, false))
Test can only be executed with a specified NS_TRAVIS

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/System/InstallationConfigurationIntegrityTest.php:75

12) SMW\Tests\System\InstallationGlobalsProviderIntegrityTest::testNamespacesWithSemanticLinksOnTravisCustomNamespace with data set #1 ('Settings', array(true, false, true, false, true, false, true, false, true, false, true, false, true, false, false, false, false, false, true, false, true, false))
Test can only be executed with a specified NS_TRAVIS

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/System/InstallationConfigurationIntegrityTest.php:75

FAILURES!
Tests: 3070, Assertions: 8797, Errors: 2, Skipped: 12.
Script php ../../tests/phpunit/phpunit.php -c phpunit.xml.dist handling the phpunit event returned with an error

**Comment by DestructionOfPlanetJupiter** [comment]
I have changed the settings for the images folder to 777 and ran the test again:

.......................................S..................... 2684 / 3071 ( 87%)
.................................................SSSSSSS..... 2745 / 3071 ( 89%)
...............................
SMW\Tests\Integration\MediaWiki\Import\Maintenance\RebuildDataMaintenanceTest testRebuildData ran for 11.115 seconds
.............................. 2806 / 3071 ( 91%)
............................................................. 2867 / 3071 ( 93%)
............................................................. 2928 / 3071 ( 95%)
............................................................. 2989 / 3071 ( 97%)
.SSS......................................................... 3050 / 3071 ( 99%)
.....................

Time: 2.1 minutes, Memory: 196.00Mb

There were 11 skipped tests:

1) SMW\Tests\Integration\MediaWiki\SearchInPageDBIntegrationTest::testSearchForGeographicCoordinateValueAsTerm
Requires 'Geographic coordinate' to be a supported data type (see Semantic Maps)

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/MediaWiki/SearchInPageDBIntegrationTest.php:62
/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/MwDBaseUnitTestCase.php:104

2) SMW\Tests\Integration\SPARQLStore\QueryResultLookupWithoutBaseStoreIntegrationTest::testQuerySubjects_afterUpdatingSemanticData
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/QueryResultLookupWithoutBaseStoreIntegrationTest.php:51

3) SMW\Tests\Integration\SPARQLStore\QueryResultLookupWithoutBaseStoreIntegrationTest::testQueryZeroResults_afterSubjectRemoval
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/QueryResultLookupWithoutBaseStoreIntegrationTest.php:51

4) SMW\Tests\Integration\SPARQLStore\QueryResultLookupWithoutBaseStoreIntegrationTest::testQuerySubjects_onNamspaceRestrictedCondition
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/QueryResultLookupWithoutBaseStoreIntegrationTest.php:51

5) SMW\Tests\Integration\SPARQLStore\QueryResultLookupWithoutBaseStoreIntegrationTest::testQuerySubobjects_afterUpdatingWithEmptyContainerAllAssociatedEntitiesGetRemovedFromGraph
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/QueryResultLookupWithoutBaseStoreIntegrationTest.php:51

6) SMW\Tests\Integration\SPARQLStore\RedirectLookupIntegrationTest::testRedirectTragetLookupForNonExistingEntry with data set #0 (SMWExpNsResource Object (...))
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/RedirectLookupIntegrationTest.php:38

7) SMW\Tests\Integration\SPARQLStore\RedirectLookupIntegrationTest::testRedirectTragetLookupForNonExistingEntry with data set #1 (SMWExpNsResource Object (...))
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/RedirectLookupIntegrationTest.php:38

8) SMW\Tests\Integration\SPARQLStore\RedirectLookupIntegrationTest::testRedirectTragetLookupForExistingEntry
Requires a SPARQLStore instance

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/Integration/SPARQLStore/RedirectLookupIntegrationTest.php:38

9) SMW\Tests\System\InstallationGlobalsProviderIntegrityTest::testNamespaceSettingOnExampleIfSet
Skip test due to missing http://example.org/id/ setting

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/System/InstallationConfigurationIntegrityTest.php:60

10) SMW\Tests\System\InstallationGlobalsProviderIntegrityTest::testNamespacesWithSemanticLinksOnTravisCustomNamespace with data set #0 ('GLOBALS', array(true, false, true, false, true, false, true, false, true, false, true, false, true, false, false, false, false, false, true, false, true, false))
Test can only be executed with a specified NS_TRAVIS

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/System/InstallationConfigurationIntegrityTest.php:75

11) SMW\Tests\System\InstallationGlobalsProviderIntegrityTest::testNamespacesWithSemanticLinksOnTravisCustomNamespace with data set #1 ('Settings', array(true, false, true, false, true, false, true, false, true, false, true, false, true, false, false, false, false, false, true, false, true, false))
Test can only be executed with a specified NS_TRAVIS

/var/www/w/rurakblue/extensions/SemanticMediaWiki/tests/phpunit/System/InstallationConfigurationIntegrityTest.php:75

OK, but incomplete, skipped, or risky tests!
Tests: 3071, Assertions: 8829, Skipped: 11.

#############
Looks fine actually.

**Comment by mwjames** [comment]
> And this is the result for:
BlueSpice activated, additional Namespace settings added (see link above) and the extensions ExtendedSearch and Universal Export deactivated:
OK, but incomplete, skipped, or risky tests!
Tests: 3071, Assertions: 8829, Skipped: 11.
>#############
Looks fine actually.

I would say that this looks rather promising, given the fact that now all tests pass which indicates an increased likelihood that both extensions are be able to coexists (except for  `ExtendedSearch and Universal Export`) .

**Comment by mwjames** [comment]
@hexmode FYI due to this discussion being about BlueSpice and SMW.

**Comment by kghbln** [comment]
@mrglaser This may be of interest for you too.

**Comment by mrglaser** [comment]
Totally of interest. There's already a fix for the namespace issue in review: https://gerrit.wikimedia.org/r/#/c/186746/ I have my team look into the ExtendedSearch/UniversalExport-thing, though.

**Comment by DestructionOfPlanetJupiter** [comment]
I have been able to deal with the NameSpace issue as outlined above. Yet, the {{#ask:}}-queries offer only empty results and the properties do not show any data or sites in which they had been used. Any idea on how to fix this?

**Comment by mwjames** [comment]
> Yet, the {{#ask:}}-queries offer only empty results and the properties do not show any data or sites in which they had been used. Any idea on how to fix this?

I can't really say from a distance but you have to know that those tests you run are covering a wide range of functionality (especially those in the `Integration` folder because those tests create pages and embedded `#ask` queries or `#set` annotations to compare the expected results against a `live` Store).

Another thing you can try out (also used during the tests) are the XML files [0, 1, 2] that by the means of `Special:Import` represent some initial data sets. If you import those XML's and still find yourself in the position of non-declared properties (or queries) then something is in vain with your setup despite the fact that the tests pass.

[0] https://github.com/SemanticMediaWiki/SemanticMediaWiki/tree/master/tests/phpunit/Integration/MediaWiki/Import/Fixtures

[1] https://github.com/SemanticMediaWiki/SemanticMediaWiki/tree/master/tests/phpunit/Benchmark/Fixtures

[2] https://gist.github.com/mwjames/2f83d3d775cbcc42b4ed

**Comment by JeroenDeDauw** [comment]
@DestructionOfPlanetJupiter pasting those long things makes it kinda hard to follow the thread here. Using pastebin or similar is nicer.

**Comment by osnard** [comment]
A fix has been merged into branches "master" and "REL1_23" of BlueSpiceExtensions repo on git.wikimedia.org: https://git.wikimedia.org/commit/mediawiki%2Fextensions%2FBlueSpiceExtensions.git/a6a6cac0f1f36d4ebfedab265e0d0f4a2564b39e

**Comment by mwjames** [comment]
@DestructionOfPlanetJupiter I'm inclined to close this issue, given to fact that from a SMW side there are no actionable items to act on.

@osnard As mentioned in a previous comment, the easiest why to test whether SMW and BlueSpice can be installed side by side is to run `composer phpunit` from the SMW root directory.

**Comment by mwjames** [comment]
Feel free to re-open this thread in case SMW can be help to solve a dedicated issue related to the BlueSpice integration.

**Comment by mwjames** [comment]
@osnard I'm not sure whether the following comment is still relevant or not, "I found the source and a bypass to the problem. When Bluespice extension "WikiAdmin" is activated, it seams to remove the namespace created by the Semantic MediaWiki extention" [0]

[0] https://www.mediawiki.org/w/index.php?title=Topic:S1ndb9v7dbw9zn0i&topic_showPostId=safhhxiqecwosu4x#flow-post-safhhxiqecwosu4x

**Comment by osnard** [comment]
@mwjames Actually the issue should be solved in current versions of BlueSpice. The link shows a posting from a year ago. I think this is not a current issue anymore.
